File: 2015/01.py
from time import time

def TrackTime(func):
    def wrapper(*a, **k):
        start = time()
        func(*a, **k)
        print("  >> in %f ms" % ((time() - start)*1000))
    return wrapper

def ParseInput(filename):
    data = []
    f = open(filename, 'r')
    data = f.read()
    f.close()
    return data

@TrackTime
def Part1(data):
    pos = 0
    for c in data:
        if c == '(': pos += 1
        if c == ')': pos -= 1
    print(f'Answer to part 1 is {pos}')

@TrackTime
def Part2(data):
    pos = 0
    res = 0
    for i, c in enumerate(data):
        if c == '(': pos += 1
        if c == ')': pos -= 1
        if pos == -1:
            res = i
            break
    print(f'Answer to part 2 is {res + 1}')


data = ParseInput('./data/01.input')
Part1(data)
Part2(data)

File: 2023/01.js
/**
 * Share your thoughts
 */

/**
 * Update year and day
 */
const year = '2023';
const day = '01';
const fs = require('fs');

const parseInput = () => {
    let filename = `./data/${day}`;
    if (process.argv.length > 2) {
        filename += '.test';
    }
    let data = fs.readFileSync(filename, 'utf8');
    return data;
}

const readCalibration = (calibration) => {
	if (calibration.length === 0) return 0;

	let first, second;
	for (let i = 0; i < calibration.length; i++) {
		if (/^\d$/.test(calibration[i])) {
			first = calibration[i];
			break;
		}
	}
	for (let i = calibration.length - 1; i >= 0; i--) {
		if (/^\d$/.test(calibration[i])) {
			second = calibration[i];
			break;
		}
	}
	return parseInt(first+second);
}

const part1 = (data) => {
    console.time('Time');
    let res = 0;

    for (const calibration of data.split('\n')) {
	    res += readCalibration(calibration);
    }

    console.log('Part 1:', res);
    console.timeEnd('Time');
}

const part2 = (data) => {
    console.time('Time');
    let res = 0;

    console.log('Part 2:', res);
    console.timeEnd('Time');
}

/**
 * Main
 */
console.log( `Advent Of Code ${year}-${day}`);
console.log('==========================');
part1(parseInput());
console.log('--------------------------');
part2(parseInput());
